tests/test_rdpg_corr.py

```diff
@@ -0,0 +1,58 @@
+import unittest
+from graspy.simulations.simulations import sample_edges, p_from_latent
+from graspy.simulations.rdpg_corr import rdpg_corr
+import numpy as np
+import pytest
+import warnings
+
+
+class Test_RDPG_Corr(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.r = 0.3
+        cls.Y = None
+        cls.X = np.random.dirichlet([20, 20], size=300)
+
+    def test_dimensions(self):
+        np.random.seed(1234)
+        A, B = rdpg_corr(
+            self.X, self.Y, self.r, rescale=False, directed=False, loops=False
+        )
+        self.assertTrue(A.shape, (300, 300))
+        self.assertTrue(B.shape, (300, 300))
+
+    def test_inputs(self):
+        x1 = np.array([[1, 1], [1, 1]])
+        x2 = np.array([[1, 1]])
+        x3 = np.zeros((2, 2, 2))
+        with self.assertRaises(TypeError):
+            p_from_latent("hi")  # wrong type
+        with self.assertRaises(ValueError):
+            p_from_latent(x1, x2)  # dimension mismatch
+        with self.assertRaises(ValueError):
+            p_from_latent(x3)  # wrong num dimensions
+        with self.assertRaises(TypeError):
+            sample_edges("XD")  # wrong type
+        with self.assertRaises(ValueError):
+            sample_edges(x3)  # wrong num dimensions
+        with self.assertRaises(ValueError):
+            sample_edges(x2)  # wrong shape for P
+
+        if any(self.X[self.X > 1]) or any(self.X[self.X < -1]):  # wrong values for P
+            raise ValueError("P values should be less than 1 and bigger than -1")
+
+    def test_rdpg_corr(self):
+        np.random.seed(123)
+        g1, g2 = rdpg_corr(
+            self.X, self.Y, self.r, rescale=False, directed=False, loops=False
+        )
+
+        # check the dimention of g1, g2
```

Comment:
And I check the dimension of g1 and g2 here to make sure they have the same dimension with P indirectly.